### PR TITLE
config: cleaner separation of tests

### PR DESCRIPTION
**Pull request by cmwaters**
This PR attempts to clearly separate testing infrastructure from the config package.

It also removes `chainID` from the `Config` struct

### CODE REVIEW

**Comment by thanethomson** [comment]
> This PR attempts to clearly separate testing infrastructure from the config package.
> 
> It also removes `chainID` from the `Config` struct

These seem like they should be two separate PRs?

**Comment by cmwaters** [comment]
> These seem like they should be two separate PRs?

Potentially although they're quite intertwined. The chainId was often used in tests

File: light/example_test.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	dbm "github.com/tendermint/tm-db"
 
 	"github.com/tendermint/tendermint/abci/example/kvstore"
+	"github.com/tendermint/tendermint/internal/test"
```
Comment (thanethomson) [inline light/example_test.go:14]: The linter seems to be complaining about this.

**Review by thanethomson — APPROVED** [review]
Other than the linter complaint, LGTM.

File: light/example_test.go (hunk 1 above)
Comment (cmwaters) [reply light/example_test.go]: Fixed 👍

**Pull request merged**